src/Native/Utils.js

```diff
@@ -16,7 +16,13 @@ function eq(rootX, rootY)
 		{
 			continue;
 		}
-		if (typeof x === 'object')
+
+		if (x instanceof Date && y instanceof Date)
+		{
+			if (x.valueOf() !== y.valueOf())
+			{ return false; }
```

Comment:
I added a test for the tempting return x.valueOf === thing. It's in a while statement, so returning true could lead to false equality for complex structures.